hydra/core/utils.py

```diff
@@ -82,17 +87,35 @@ def filter_overrides(overrides: Sequence[str]) -> Sequence[str]:
     return [x for x in overrides if not x.startswith("hydra.")]
 
 
+def _get_callbacks_for_run_job(hydra_context: Optional[HydraContext]) -> "Callbacks":
+    if hydra_context is None:
+        warnings.warn(
+            message=dedent(
+                """
+            run_job's signature has changed in Hydra 1.1.
+            Support for the old style will be removed in Hydra 1.2.
+            For more info, check https://github.com/facebookresearch/hydra/pull/1581.
+            """
+            ),
```

Comment:
Add a comment like `# DEPRECATED: remove in 1.2` like in other places we are deprecating things.
You can also add some info in the comment about he plan (make hydra_context non-optional).